### PR TITLE
create and dispatch OnLinkParsedEvent

### PR DESCRIPTION
**Pull request by nikophil**
Hello,

we would like to create a command which tests all external urls response code.
To achieve that goal, we need another hook point from which we can grab the link's url and the current file being parsed.

thanks.

### CODE REVIEW

File: lib/Event/OnLinkParsedEvent.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+final class OnLinkParsedEvent extends EventArgs
+{
+    public const ON_LINK_PARSED = 'onLinkParsed';
+
+    /** @var string */
+    private $url;
+    /** @var string */
+    private $linkType;
+    /** @var string */
+    private $currentFileName;
```
Comment (weaverryan) [inline lib/Event/OnLinkParsedEvent.php]: I'm not positive about the Doctrine coding standards, but my guess is that the phpdoc above these is not needed/wanted

Comment (jwage) [reply lib/Event/OnLinkParsedEvent.php]: They are required so phpstan, etc knows what type the properties are so we can enforce type consistency.

Comment (jwage) [reply lib/Event/OnLinkParsedEvent.php]: We also need a new line inbetween each property.

Comment (nikophil) [reply lib/Event/OnLinkParsedEvent.php]: done ;)

**Pull request merged**